code/lib/preview-api/src/modules/preview-web/docs-context/DocsContext.ts

```diff
@@ -71,13 +72,18 @@ export class DocsContext<TRenderer extends Renderer> implements DocsContextProps
     if (!this.exportsToCSFFile.has(csfFile.moduleExports)) {
       throw new Error('Cannot attach a CSF file that has not been referenced');
     }
+    if (this.attachedCSFFiles.has(csfFile)) {
+      // this CSF file is already attached, don't do anything
+      return;
+    }
```

Comment:
This is the actual fix for the issue, if we actually don't think supporting multiple attached CSF file is a valid use case then this PR can be reduced to these lines.